### PR TITLE
Automatically install ds_ctcdecoder in setup.py

### PR DESCRIPTION
**Pull request by reuben**

### CODE REVIEW

File: taskcluster/tc-transfer-tests.sh [hunk 1]
```diff
@@ -15,16 +15,12 @@ virtualenv_activate "${pyalias}" "deepspeech"
...
 popd
 set +o pipefail
 
-pushd ${HOME}/DeepSpeech/ds/
-    verify_ctcdecoder_url
-popd
-
 decoder_pkg_url=$(get_python_pkg_url ${pyver_pkg} ${py_unicode_type} "ds_ctcdecoder" "${DECODER_ARTIFACTS_ROOT}")
-LD_LIBRARY_PATH=${PY37_LDPATH}:$LD_LIBRARY_PATH pip install --verbose --only-binary :all: ${PY37_SOURCE_PACKAGE} ${decoder_pkg_url} | cat
+LD_LIBRARY_PATH=${PY37_LDPATH}:$LD_LIBRARY_PATH pip install --upgrade --verbose --only-binary :all: ${PY37_SOURCE_PACKAGE} ${decoder_pkg_url} | cat
```
Comment (reuben) [inline taskcluster/tc-transfer-tests.sh]: this is a spurious change, I'll remove it

File: setup.py [hunk 2]
```diff
@@ -1,13 +1,85 @@
...
+
+    if plat == 'darwin':
+        plat = 'macosx_10_10'
+
+    is_ucs2 = sys.maxunicode < 0x10ffff
+    m_or_mu = 'mu' if is_ucs2 else 'm'
+
+    pyver = ''.join(str(i) for i in sys.version_info[0:2])
+
+    return 'ds_ctcdecoder @ https://community-tc.services.mozilla.com/api/index/v1/task/project.deepspeech.deepspeech.native_client.{branch_name}.{tc_arch_string}/artifacts/public/ds_ctcdecoder-{ds_version}-cp{pyver}-cp{pyver}{m_or_mu}-{platform}_{arch}.whl'.format(
```
Comment (lissyx) [inline setup.py]: We could likely get the root URL from `DECODER_ARTIFACTS_ROOT` in CI and this way it should work for all cases, even bumping version.

File: taskcluster/tc-all-utils.sh [hunk 3]
```diff
@@ -122,25 +122,3 @@ verify_bazel_rebuild()
     exit 1
   fi;
 }
-
-# Should be called from context where Python virtualenv is set
-verify_ctcdecoder_url()
```
Comment (lissyx) [inline taskcluster/tc-all-utils.sh]: Maybe worth adding unit test coverage of the similar feature we have in `setup.py` ? Your call.

File: doc/TRAINING.rst [hunk 4]
```diff
@@ -46,22 +46,15 @@ Install the required dependencies using ``pip3``\ :
 .. code-block:: bash
 
    cd DeepSpeech
-   pip3 install -e .
+   pip3 install --upgrade pip wheel setuptools
```
Comment (lissyx) [inline doc/TRAINING.rst]: Do we need latest version ?

Comment (reuben) [reply doc/TRAINING.rst]: I don't think so, but depending on the OS/distro/Python version fresh virtualenvs come with older versions so for users of the training code I added this instruction to be safe.

Comment (reuben) [reply doc/TRAINING.rst]: For example some versions are incompatible with TF 1.15 wheels.

Comment (lissyx) [reply doc/TRAINING.rst]: Sure, but forcing always latest version can bring its own issues. I'd feel better if we could stick to well known versions, like we do at several other places.

File: taskcluster/tc-all-utils.sh (hunk 3 above)
Comment (reuben) [reply taskcluster/tc-all-utils.sh]: I'm not sure how to test `setup.py`. I guess installing the package and then trying to import the decoder dependency to see if it works?

Comment (reuben) [reply taskcluster/tc-all-utils.sh]: It's still not going to test the entire URL computation part that users will use, though.

**Pull request merged**